[CI] GroupByOptimizerTests testOrderByType failing

Looks like this just and a couple of other tests in this suite have started failing on Windows. Muting.

**Build scan:**
https://gradle-enterprise.elastic.co/s/dy2pa5e64ypqu/tests/:x-pack:plugin:transform:test/org.elasticsearch.xpack.transform.transforms.pivot.GroupByOptimizerTests/testOrderByType


**Reproduction line:**
```
gradlew ':x-pack:plugin:transform:test' --tests "org.elasticsearch.xpack.transform.transforms.pivot.GroupByOptimizerTests.testOrderByType" -Dtests.seed=DCCD03661C2C4DDB -Dtests.locale=uk -Dtests.timezone=America/Creston -Druntime.java=21
```

**Applicable branches:**
main

**Reproduces locally?:**
Didn't try

**Failure history:**
[Failure dashboard for `org.elasticsearch.xpack.transform.transforms.pivot.GroupByOptimizerTests#testOrderByType`](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('testOrderByType'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.xpack.transform.transforms.pivot.GroupByOptimizerTests'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))


**Failure excerpt:**
```
java.lang.AssertionError: 
Expected: an empty collection
     but: <[LEAK: resource was not cleaned up before it was garbage-collected.
Recent access records: 
Created at:
	in [elasticsearch[testHandleBulkResponseWithNoIrrecoverableFailures][generic][T#1]][testHandleBulkResponseWithNoIrrecoverableFailures]
	org.elasticsearch.action.search.SearchResponse.<init>(SearchResponse.java:88)
	org.elasticsearch.action.search.SearchResponse.<init>(SearchResponse.java:133)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests.lambda$returnHit$26(TransformIndexerFailureHandlingTests.java:1001)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests$MockedTransformIndexer.doNextSearch(TransformIndexerFailureHandlingTests.java:199)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.triggerNextSearch(AsyncTwoPhaseIndexer.java:623)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.nextSearch(AsyncTwoPhaseIndexer.java:610)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.lambda$maybeTriggerAsyncJob$4(AsyncTwoPhaseIndexer.java:233)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$6(TransformIndexer.java:301)
	org.elasticsearch.action.ActionListenerImplementations$ResponseWrappingActionListener.onResponse(ActionListenerImplementations.java:245)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$7(TransformIndexer.java:333)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.common.AbstractCompositeAggFunction.getInitialProgressFromResponse(AbstractCompositeAggFunction.java:206)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$10(TransformIndexer.java:330)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.action.ActionListener.respondAndRelease(ActionListener.java:366)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests$MockedTransformIndexer.doGetInitialProgress(TransformIndexerFailureHandlingTests.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$13(TransformIndexer.java:329)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests$MockedTransformIndexer.createCheckpoint(TransformIndexerFailureHandlingTests.java:186)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$14(TransformIndexer.java:311)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$16(TransformIndexer.java:373)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests$MockedTransformIndexer.doGetFieldMappings(TransformIndexerFailureHandlingTests.java:287)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$18(TransformIndexer.java:380)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$21(TransformIndexer.java:418)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.onStart(TransformIndexer.java:450)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.lambda$maybeTriggerAsyncJob$5(AsyncTwoPhaseIndexer.java:230)
	org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:917)
	java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	java.base/java.lang.Thread.run(Thread.java:1583), LEAK: resource was not cleaned up before it was garbage-collected.
Recent access records: 
Created at:
	in [elasticsearch[testHandleBulkResponseWithIrrecoverableFailures][generic][T#1]][testHandleBulkResponseWithIrrecoverableFailures]
	org.elasticsearch.action.search.SearchResponse.<init>(SearchResponse.java:88)
	org.elasticsearch.action.search.SearchResponse.<init>(SearchResponse.java:133)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests.lambda$returnHit$26(TransformIndexerFailureHandlingTests.java:1001)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests$MockedTransformIndexer.doNextSearch(TransformIndexerFailureHandlingTests.java:199)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.triggerNextSearch(AsyncTwoPhaseIndexer.java:623)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.nextSearch(AsyncTwoPhaseIndexer.java:610)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.lambda$maybeTriggerAsyncJob$4(AsyncTwoPhaseIndexer.java:233)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$6(TransformIndexer.java:301)
	org.elasticsearch.action.ActionListenerImplementations$ResponseWrappingActionListener.onResponse(ActionListenerImplementations.java:245)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$7(TransformIndexer.java:333)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.common.AbstractCompositeAggFunction.getInitialProgressFromResponse(AbstractCompositeAggFunction.java:206)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$10(TransformIndexer.java:330)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.action.ActionListener.respondAndRelease(ActionListener.java:366)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests$MockedTransformIndexer.doGetInitialProgress(TransformIndexerFailureHandlingTests.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$13(TransformIndexer.java:329)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests$MockedTransformIndexer.createCheckpoint(TransformIndexerFailureHandlingTests.java:186)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$14(TransformIndexer.java:311)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$16(TransformIndexer.java:373)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests$MockedTransformIndexer.doGetFieldMappings(TransformIndexerFailureHandlingTests.java:287)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$18(TransformIndexer.java:380)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$21(TransformIndexer.java:418)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.onStart(TransformIndexer.java:450)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.lambda$maybeTriggerAsyncJob$5(AsyncTwoPhaseIndexer.java:230)
	org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:917)
	java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	java.base/java.lang.Thread.run(Thread.java:1583), LEAK: resource was not cleaned up before it was garbage-collected.
Recent access records: 
Created at:
	in [elasticsearch[testHandleBulkResponseWithNoFailures][generic][T#1]][testHandleBulkResponseWithNoFailures]
	org.elasticsearch.action.search.SearchResponse.<init>(SearchResponse.java:88)
	org.elasticsearch.action.search.SearchResponse.<init>(SearchResponse.java:133)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests.lambda$returnHit$26(TransformIndexerFailureHandlingTests.java:1001)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests$MockedTransformIndexer.doNextSearch(TransformIndexerFailureHandlingTests.java:199)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.triggerNextSearch(AsyncTwoPhaseIndexer.java:623)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.nextSearch(AsyncTwoPhaseIndexer.java:610)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.lambda$maybeTriggerAsyncJob$4(AsyncTwoPhaseIndexer.java:233)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$6(TransformIndexer.java:301)
	org.elasticsearch.action.ActionListenerImplementations$ResponseWrappingActionListener.onResponse(ActionListenerImplementations.java:245)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$7(TransformIndexer.java:333)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.common.AbstractCompositeAggFunction.getInitialProgressFromResponse(AbstractCompositeAggFunction.java:206)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$10(TransformIndexer.java:330)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.action.ActionListener.respondAndRelease(ActionListener.java:366)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests$MockedTransformIndexer.doGetInitialProgress(TransformIndexerFailureHandlingTests.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$13(TransformIndexer.java:329)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests$MockedTransformIndexer.createCheckpoint(TransformIndexerFailureHandlingTests.java:186)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$14(TransformIndexer.java:311)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$16(TransformIndexer.java:373)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests$MockedTransformIndexer.doGetFieldMappings(TransformIndexerFailureHandlingTests.java:287)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$18(TransformIndexer.java:380)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$21(TransformIndexer.java:418)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.onStart(TransformIndexer.java:450)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.lambda$maybeTriggerAsyncJob$5(AsyncTwoPhaseIndexer.java:230)
	org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:917)
	java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	java.base/java.lang.Thread.run(Thread.java:1583), LEAK: resource was not cleaned up before it was garbage-collected.
Recent access records: 
Created at:
	in [elasticsearch[testHandleBulkResponseWithNoFailures][generic][T#1]][testHandleBulkResponseWithNoFailures]
	org.elasticsearch.action.search.SearchResponse.<init>(SearchResponse.java:88)
	org.elasticsearch.action.search.SearchResponse.<init>(SearchResponse.java:133)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests.lambda$returnHit$26(TransformIndexerFailureHandlingTests.java:1001)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests$MockedTransformIndexer.doNextSearch(TransformIndexerFailureHandlingTests.java:199)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.triggerNextSearch(AsyncTwoPhaseIndexer.java:623)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.nextSearch(AsyncTwoPhaseIndexer.java:610)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.onBulkResponse(AsyncTwoPhaseIndexer.java:581)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.lambda$onSearchResponse$11(AsyncTwoPhaseIndexer.java:547)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.ClientTransformIndexer.handleBulkResponse(ClientTransformIndexer.java:187)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests$MockedTransformIndexer.doNextBulk(TransformIndexerFailureHandlingTests.java:217)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.onSearchResponse(AsyncTwoPhaseIndexer.java:533)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests$MockedTransformIndexer.doNextSearch(TransformIndexerFailureHandlingTests.java:200)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.triggerNextSearch(AsyncTwoPhaseIndexer.java:623)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.nextSearch(AsyncTwoPhaseIndexer.java:610)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.lambda$maybeTriggerAsyncJob$4(AsyncTwoPhaseIndexer.java:233)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$6(TransformIndexer.java:301)
	org.elasticsearch.action.ActionListenerImplementations$ResponseWrappingActionListener.onResponse(ActionListenerImplementations.java:245)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$7(TransformIndexer.java:333)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.common.AbstractCompositeAggFunction.getInitialProgressFromResponse(AbstractCompositeAggFunction.java:206)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$10(TransformIndexer.java:330)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.action.ActionListener.respondAndRelease(ActionListener.java:366)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests$MockedTransformIndexer.doGetInitialProgress(TransformIndexerFailureHandlingTests.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$13(TransformIndexer.java:329)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests$MockedTransformIndexer.createCheckpoint(TransformIndexerFailureHandlingTests.java:186)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$14(TransformIndexer.java:311)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$16(TransformIndexer.java:373)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests$MockedTransformIndexer.doGetFieldMappings(TransformIndexerFailureHandlingTests.java:287)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$18(TransformIndexer.java:380)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$21(TransformIndexer.java:418)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.onStart(TransformIndexer.java:450)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.lambda$maybeTriggerAsyncJob$5(AsyncTwoPhaseIndexer.java:230)
	org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:917)
	java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	java.base/java.lang.Thread.run(Thread.java:1583), LEAK: resource was not cleaned up before it was garbage-collected.
Recent access records: 
Created at:
	in [elasticsearch[testHandleBulkResponseWithIndexNotFound][generic][T#1]][testHandleBulkResponseWithIndexNotFound]
	org.elasticsearch.action.search.SearchResponse.<init>(SearchResponse.java:88)
	org.elasticsearch.action.search.SearchResponse.<init>(SearchResponse.java:133)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests.lambda$returnHit$26(TransformIndexerFailureHandlingTests.java:1001)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests$MockedTransformIndexer.doNextSearch(TransformIndexerFailureHandlingTests.java:199)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.triggerNextSearch(AsyncTwoPhaseIndexer.java:623)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.nextSearch(AsyncTwoPhaseIndexer.java:610)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.lambda$maybeTriggerAsyncJob$4(AsyncTwoPhaseIndexer.java:233)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$6(TransformIndexer.java:301)
	org.elasticsearch.action.ActionListenerImplementations$ResponseWrappingActionListener.onResponse(ActionListenerImplementations.java:245)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$7(TransformIndexer.java:333)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.common.AbstractCompositeAggFunction.getInitialProgressFromResponse(AbstractCompositeAggFunction.java:206)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$10(TransformIndexer.java:330)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.action.ActionListener.respondAndRelease(ActionListener.java:366)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests$MockedTransformIndexer.doGetInitialProgress(TransformIndexerFailureHandlingTests.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$13(TransformIndexer.java:329)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests$MockedTransformIndexer.createCheckpoint(TransformIndexerFailureHandlingTests.java:186)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$14(TransformIndexer.java:311)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$16(TransformIndexer.java:373)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests$MockedTransformIndexer.doGetFieldMappings(TransformIndexerFailureHandlingTests.java:287)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$18(TransformIndexer.java:380)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$21(TransformIndexer.java:418)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.onStart(TransformIndexer.java:450)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.lambda$maybeTriggerAsyncJob$5(AsyncTwoPhaseIndexer.java:230)
	org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:917)
	java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	java.base/java.lang.Thread.run(Thread.java:1583)]>

  at __randomizedtesting.SeedInfo.seed([DCCD03661C2C4DDB:8F74AEAB03908C6C]:0)
  at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
  at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:6)
  at org.elasticsearch.test.ESTestCase.assertThat(ESTestCase.java:2167)
  at org.elasticsearch.test.ESTestCase.checkStaticState(ESTestCase.java:736)
  at org.elasticsearch.test.ESTestCase.after(ESTestCase.java:527)
  at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
  at java.lang.reflect.Method.invoke(Method.java:580)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:1004)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.junit.rules.RunRules.evaluate(RunRules.java:20)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:48)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.junit.rules.RunRules.evaluate(RunRules.java:20)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at org.junit.rules.RunRules.evaluate(RunRules.java:20)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
  at java.lang.Thread.run(Thread.java:1583)

```

--- a/x-pack/plugin/transform/src/test/java/org/elasticsearch/xpack/transform/transforms/TransformIndexerFailureHandlingTests.java
+++ b/x-pack/plugin/transform/src/test/java/org/elasticsearch/xpack/transform/transforms/TransformIndexerFailureHandlingTests.java
@@ -535,7 +535,10 @@ public class TransformIndexerFailureHandlingTests extends ESTestCase {
         );
         try {
             AtomicReference<IndexerState> state = new AtomicReference<>(IndexerState.STOPPED);
-            Function<SearchRequest, SearchResponse> searchFunction = searchRequest -> searchResponse;
+            Function<SearchRequest, SearchResponse> searchFunction = searchRequest -> {
+                searchResponse.mustIncRef();
+                return searchResponse;
+            };
 
             Function<BulkRequest, BulkResponse> bulkFunction = bulkRequest -> new BulkResponse(new BulkItemResponse[0], 100);
 
@@ -627,7 +630,10 @@ public class TransformIndexerFailureHandlingTests extends ESTestCase {
         );
         try {
             AtomicReference<IndexerState> state = new AtomicReference<>(IndexerState.STOPPED);
-            Function<SearchRequest, SearchResponse> searchFunction = searchRequest -> searchResponse;
+            Function<SearchRequest, SearchResponse> searchFunction = searchRequest -> {
+                searchResponse.mustIncRef();
+                return searchResponse;
+            };
 
             Function<BulkRequest, BulkResponse> bulkFunction = bulkRequest -> new BulkResponse(new BulkItemResponse[0], 100);
 
@@ -735,6 +741,7 @@ public class TransformIndexerFailureHandlingTests extends ESTestCase {
                             new ShardSearchFailure[] { new ShardSearchFailure(new Exception()) }
                         );
                     }
+                    searchResponse.mustIncRef();
                     return searchResponse;
                 }
             };

--- a/x-pack/plugin/transform/src/test/java/org/elasticsearch/xpack/transform/transforms/TransformIndexerFailureHandlingTests.java
+++ b/x-pack/plugin/transform/src/test/java/org/elasticsearch/xpack/transform/transforms/TransformIndexerFailureHandlingTests.java
@@ -21,7 +21,6 @@ import org.elasticsearch.action.search.SearchPhaseExecutionException;
 import org.elasticsearch.action.search.SearchRequest;
 import org.elasticsearch.action.search.SearchResponse;
 import org.elasticsearch.action.search.ShardSearchFailure;
-import org.elasticsearch.client.internal.Client;
 import org.elasticsearch.client.internal.ParentTaskAssigningClient;
 import org.elasticsearch.cluster.metadata.IndexNameExpressionResolver;
 import org.elasticsearch.cluster.service.ClusterService;
@@ -39,7 +38,6 @@ import org.elasticsearch.search.SearchHits;
 import org.elasticsearch.search.profile.SearchProfileResults;
 import org.elasticsearch.search.suggest.Suggest;
 import org.elasticsearch.test.ESTestCase;
-import org.elasticsearch.test.client.NoOpClient;
 import org.elasticsearch.threadpool.ThreadPool;
 import org.elasticsearch.xpack.core.common.notifications.Level;
 import org.elasticsearch.xpack.core.indexing.IndexerState;
@@ -105,7 +103,6 @@ import static org.mockito.Mockito.mock;
  */
 public class TransformIndexerFailureHandlingTests extends ESTestCase {
 
-    private Client client;
     private ThreadPool threadPool;
     private static final Function<BulkRequest, BulkResponse> EMPTY_BULK_RESPONSE = bulkRequest -> new BulkResponse(
         new BulkItemResponse[0],
@@ -127,7 +124,6 @@ public class TransformIndexerFailureHandlingTests extends ESTestCase {
             ClusterService clusterService,
             IndexNameExpressionResolver indexNameExpressionResolver,
             TransformExtension transformExtension,
-            String executorName,
             IndexBasedTransformConfigManager transformsConfigManager,
             CheckpointProvider checkpointProvider,
             TransformConfig transformConfig,
@@ -195,12 +191,7 @@ public class TransformIndexerFailureHandlingTests extends ESTestCase {
                 throw new IllegalStateException(e);
             }
 
-            try {
-                SearchResponse response = searchFunction.apply(buildSearchRequest().v2());
-                nextPhase.onResponse(response);
-            } catch (Exception e) {
-                nextPhase.onFailure(e);
-            }
+            ActionListener.run(nextPhase, l -> ActionListener.respondAndRelease(l, searchFunction.apply(buildSearchRequest().v2())));
         }
 
         @Override
@@ -307,7 +298,6 @@ public class TransformIndexerFailureHandlingTests extends ESTestCase {
     @Before
     public void setUpMocks() {
         threadPool = createThreadPool();
-        client = new NoOpClient(threadPool);
     }
 
     @After
@@ -349,17 +339,7 @@ public class TransformIndexerFailureHandlingTests extends ESTestCase {
         TransformAuditor auditor = MockTransformAuditor.createMockAuditor();
         TransformContext context = new TransformContext(TransformTaskState.STARTED, "", 0, mock(TransformContext.Listener.class));
 
-        MockedTransformIndexer indexer = createMockIndexer(
-            config,
-            state,
-            searchFunction,
-            bulkFunction,
-            null,
-            threadPool,
-            ThreadPool.Names.GENERIC,
-            auditor,
-            context
-        );
+        MockedTransformIndexer indexer = createMockIndexer(config, state, searchFunction, bulkFunction, null, threadPool, auditor, context);
         final CountDownLatch latch = indexer.newLatch(1);
         indexer.start();
         assertThat(indexer.getState(), equalTo(IndexerState.STARTED));
@@ -439,7 +419,6 @@ public class TransformIndexerFailureHandlingTests extends ESTestCase {
                 bulkFunction,
                 null,
                 threadPool,
-                ThreadPool.Names.GENERIC,
                 auditor,
                 context
             );
@@ -500,17 +479,7 @@ public class TransformIndexerFailureHandlingTests extends ESTestCase {
         TransformContext.Listener contextListener = createContextListener(failIndexerCalled, failureMessage);
         TransformContext context = new TransformContext(TransformTaskState.STARTED, "", 0, contextListener);
 
-        MockedTransformIndexer indexer = createMockIndexer(
-            config,
-            state,
-            searchFunction,
-            bulkFunction,
-            null,
-            threadPool,
-            ThreadPool.Names.GENERIC,
-            auditor,
-            context
-        );
+        MockedTransformIndexer indexer = createMockIndexer(config, state, searchFunction, bulkFunction, null, threadPool, auditor, context);
 
         final CountDownLatch latch = indexer.newLatch(1);
 
@@ -595,7 +564,6 @@ public class TransformIndexerFailureHandlingTests extends ESTestCase {
                 bulkFunction,
                 deleteByQueryFunction,
                 threadPool,
-                ThreadPool.Names.GENERIC,
                 auditor,
                 context
             );
@@ -694,7 +662,6 @@ public class TransformIndexerFailureHandlingTests extends ESTestCase {
                 bulkFunction,
                 deleteByQueryFunction,
                 threadPool,
-                ThreadPool.Names.GENERIC,
                 auditor,
                 context
             );
@@ -788,7 +755,6 @@ public class TransformIndexerFailureHandlingTests extends ESTestCase {
                 bulkFunction,
                 null,
                 threadPool,
-                ThreadPool.Names.GENERIC,
                 auditor,
                 context
             );
@@ -889,17 +855,7 @@ public class TransformIndexerFailureHandlingTests extends ESTestCase {
             )
         );
 
-        MockedTransformIndexer indexer = createMockIndexer(
-            config,
-            state,
-            searchFunction,
-            bulkFunction,
-            null,
-            threadPool,
-            ThreadPool.Names.GENERIC,
-            auditor,
-            context
-        );
+        MockedTransformIndexer indexer = createMockIndexer(config, state, searchFunction, bulkFunction, null, threadPool, auditor, context);
 
         indexer.handleFailure(
             new SearchPhaseExecutionException(
@@ -1056,7 +1012,6 @@ public class TransformIndexerFailureHandlingTests extends ESTestCase {
             bulkFunction,
             null,
             threadPool,
-            ThreadPool.Names.GENERIC,
             mock(TransformAuditor.class),
             new TransformContext(TransformTaskState.STARTED, "", 0, listener),
             1
@@ -1166,17 +1121,7 @@ public class TransformIndexerFailureHandlingTests extends ESTestCase {
             )
         );
 
-        MockedTransformIndexer indexer = createMockIndexer(
-            config,
-            state,
-            searchFunction,
-            bulkFunction,
-            null,
-            threadPool,
-            ThreadPool.Names.GENERIC,
-            auditor,
-            context
-        );
+        MockedTransformIndexer indexer = createMockIndexer(config, state, searchFunction, bulkFunction, null, threadPool, auditor, context);
 
         for (int i = 0; i < expectedEffectiveNumFailureRetries; ++i) {
             indexer.handleFailure(new Exception("exception no. " + (i + 1)));
@@ -1209,22 +1154,10 @@ public class TransformIndexerFailureHandlingTests extends ESTestCase {
         Function<BulkRequest, BulkResponse> bulkFunction,
         Function<DeleteByQueryRequest, BulkByScrollResponse> deleteByQueryFunction,
         ThreadPool threadPool,
-        String executorName,
         TransformAuditor auditor,
         TransformContext context
     ) {
-        return createMockIndexer(
-            config,
-            state,
-            searchFunction,
-            bulkFunction,
-            deleteByQueryFunction,
-            threadPool,
-            executorName,
-            auditor,
-            context,
-            0
-        );
+        return createMockIndexer(config, state, searchFunction, bulkFunction, deleteByQueryFunction, threadPool, auditor, context, 0);
     }
 
     private MockedTransformIndexer createMockIndexer(
@@ -1234,7 +1167,6 @@ public class TransformIndexerFailureHandlingTests extends ESTestCase {
         Function<BulkRequest, BulkResponse> bulkFunction,
         Function<DeleteByQueryRequest, BulkByScrollResponse> deleteByQueryFunction,
         ThreadPool threadPool,
-        String executorName,
         TransformAuditor auditor,
         TransformContext context,
         int doProcessCount
@@ -1250,7 +1182,6 @@ public class TransformIndexerFailureHandlingTests extends ESTestCase {
             mock(ClusterService.class),
             mock(IndexNameExpressionResolver.class),
             mock(TransformExtension.class),
-            executorName,
             transformConfigManager,
             mock(CheckpointProvider.class),
             config,

--- a/x-pack/plugin/transform/src/test/java/org/elasticsearch/xpack/transform/transforms/pivot/GroupByOptimizerTests.java
+++ b/x-pack/plugin/transform/src/test/java/org/elasticsearch/xpack/transform/transforms/pivot/GroupByOptimizerTests.java
@@ -7,7 +7,6 @@
 
 package org.elasticsearch.xpack.transform.transforms.pivot;
 
-import org.apache.lucene.tests.util.LuceneTestCase.AwaitsFix;
 import org.elasticsearch.test.ESTestCase;
 import org.elasticsearch.xpack.core.transform.TransformConfigVersion;
 import org.elasticsearch.xpack.core.transform.transforms.pivot.SingleGroupSource;
@@ -28,7 +27,6 @@ import static org.elasticsearch.xpack.core.transform.transforms.pivot.HistogramG
 import static org.elasticsearch.xpack.core.transform.transforms.pivot.TermsGroupSourceTests.randomTermsGroupSource;
 import static org.elasticsearch.xpack.core.transform.transforms.pivot.TermsGroupSourceTests.randomTermsGroupSourceNoScript;
 
-@AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/108530")
 public class GroupByOptimizerTests extends ESTestCase {
 
     public void testOneGroupBy() {

